symbolic-base/src/ZkFold/Base/Data/Logical.hs

```diff
@@ -0,0 +1,220 @@
+{-# LANGUAGE DerivingVia          #-}
+{-# LANGUAGE TypeOperators        #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module ZkFold.Base.Data.Logical
+  ( Boolean (..)
+  , all, any, and, or
+  , Conditional (..)
+  , ifThenElse
+  , Eq (..)
+  , elem
+  , GConditional (..)
+  , GEq (..)
+  ) where
+
+import           Control.Category
+import qualified Data.Bool                        as H
+import           Data.Foldable                    hiding (all, and, any, elem, or)
+import           Data.Functor.Identity
+import qualified Data.Functor.Rep                 as R
+import           Data.Kind
+import qualified GHC.Generics                     as G
+import           Prelude                          (type (~))
+import qualified Prelude                          as H
+
+import           ZkFold.Base.Algebra.Basic.Number
+import           ZkFold.Base.Data.Vector
+
+class Boolean b where
```

Comment:
I think some documentation describing the laws of the class could be useful here